packages/hardhat/test/e2e/periphery/controller-helper.ts

```diff
@@ -1356,4 +1460,635 @@ it("Close vault LP and open new one-siced LP with just oSQTH ", async () => {
+    console.log('vaultAfter.collateralAmount', vaultAfter.collateralAmount.toString())
+    console.log('vaultAfter.shortAmount', vaultAfter.shortAmount.toString())
+    console.log('wethAmountToLp',wethAmountToLp.toString())
+    console.log('flashloanAmount',flashLoanAmount.toString())
+    console.log('wPowerPerpAmountInLPBefore', wPowerPerpAmountInLPBefore.toString())
+    console.log('wPowerPerpAmountToLp', wPowerPerpAmountToLp.toString())
+    console.log('depositorEthDiff', depositorEthDiff.toString())
+    console.log('depositorSqueethDiff', depositorSqueethDiff.toString())
+    console.log('gasSpent', gasSpent.toString())
+
+    // expect(positionAfter.tickLower === newTickLower).to.be.true
+    expect(vaultSqueethDiff.sub(wPowerPerpAmountToLp).eq(0)).to.be.true
+    // LP eth comes from vault
+    expect(vaultEthDiff.add(lpEthDiff).lte(10)).to.be.true
+    // Nft id unchanged
+    expect(vaultAfter.NftCollateralId==vaultBefore.NftCollateralId).to.be.true
+    // Squeeth convervation
+    expect(lpSqueethDiff.sub(vaultSqueethDiff).add(depositorSqueethDiff).abs().lte(10)).to.be.true
+    })
+    })
+
+
+    describe("Rebalance LP in vault: reduce oSQTH < debt, burn, withdraw >0 eth", async () => {
+      before("Mint new full range LP outside of vault" , async () => {
+        // Mint 50 squeeth in new vault
+        const normFactor = await controller.getExpectedNormalizationFactor()
+        const mintWSqueethAmount = ethers.utils.parseUnits('50')
+        const mintRSqueethAmount = mintWSqueethAmount.mul(normFactor).div(one)
+        const ethPrice = await oracle.getTwap(ethUsdcPool.address, weth.address, usdc.address, 420, true)
+        const scaledEthPrice = ethPrice.div(10000)
+        const debtInEth = mintRSqueethAmount.mul(scaledEthPrice).div(one)
+        const collateralAmount = debtInEth.mul(3).div(2).add(ethers.utils.parseUnits('0.01'))
+        const squeethPrice = await oracle.getTwap(wSqueethPool.address, wSqueeth.address, weth.address, 1, true)
+        const collateralToLp = mintWSqueethAmount.mul(squeethPrice).div(one)
+        console.log('collateralToLp', collateralToLp.toString())
+        console.log('collateralAmount', collateralAmount.toString())
+        console.log('mintWSqueethAmount', mintWSqueethAmount.toString())
+        const batchMintLpParams = {
+          recipient: depositor.address,
+          wPowerPerpPool: wSqueethPool.address,
+          vaultId: 0,
+          wPowerPerpAmount: mintWSqueethAmount,
+          collateralToDeposit: collateralAmount,
+          collateralToLp: collateralToLp,
+          amount0Min: 0,
+          amount1Min: 0,
+          lowerTick: -887220,
+          upperTick: 887220
+        }
+        // Batch mint new full range LP
+        await controllerHelper.connect(depositor).batchMintLp(batchMintLpParams, {value: collateralAmount.add(collateralToLp)});
+       })
+    
+        
+    it("Reduce oSQTH, burn, withdraw >0 eth", async () => {
```

Comment:
this sells oSQTH, then mints oSQTH and deposits ETH, right? no burning or withdrawing

Reply:
changed